pkg/services/live/live.go

```diff
@@ -272,14 +277,28 @@ func (g *GrafanaLive) initChannel(id ChannelIdentifier) (models.ChannelHandler,
 			h := &PluginHandler{
 				Plugin: p,
 			}
-			return h.GetHandlerForPath(id.Path)
+			return h.GetHandlerForPath(id.Path, g.Publish)
 		}
 		return nil, fmt.Errorf("unknown plugin: %s", id.Namespace)
 	}
 
 	return nil, fmt.Errorf("invalid scope: %s", id.Scope)
 }
 
+var handlerMu = sync.Mutex{}
+
+func RegisterHandler(datasourceID string, channelHandlerProvider models.ChannelHandlerProvider) error {
```

Comment:
I wonder if this could be a method on GrafanaLive instead of free function, so it can avoid using global variables?

Reply:
This was initially what I wanted to do, but since `Live` is a property of the HTTP server, I wasn't sure how best to expose it so that it can be used, for example, by the CloudWatch data source

Reply:
Hmm.. I guess it should be looked into.

Reply:
Yeah, I'll discuss with @ryantxu . Definitely agree that having global variables isn't ideal

Reply:
ideally the datasource handlers are tied to their instance, not registered ad-hoc.

Reply:
@aknuds1 this will require routing that does not yet exist.  Although centrifuge lets us publish data to any channel, we want to add more clear constraints and structure so that portions of the channel namespace are "owend" by various parts of the system:

The thee scopes we have defined (but not yet implemented)

* /live/grafana/${feature}
* /live/ds/${datasourceId | name?}
* /live/plugin/${pluginId}

This PR now uses hardcodes its access to `plugin/cloudwatch/*`  For the particular approach in this PR (start a runner from a query, and listen to results on a channel) that is totally fine and can be improved without issue.